ansible/roles/nginx/templates/etsin_nginx.conf

```diff
@@ -193,4 +193,14 @@ http {
             proxy_set_header X-Forwarded-Proto $scheme;
         }
     }
+
+    # Redirect from etsin.avointiede.fi (old Etsin) to etsin.fairdata.fi (new Etsin)
+    server {
+        server_name etsin.avointiede.fi;
+        listen 443 ssl;
+        ssl_certificate {{ ssl_certificates_path }}/etsin.avointiede.fi.crt;
+        ssl_certificate_key {{ ssl_certificates_path }}/etsin.avointiede.fi.key;
+
+        return 301 https://etsin.fairdata.fi$request_uri;
+    }
```

Comment:
Did you test if this redirect works and the used ssl-related parameters are enough? In the other ssl-server blocks, there are quite a few more parameters used.

Otherwise the block looks like it could certainly work. It could be improved a lot by parameterizing server_name, the cert names, and the redirect url. This way the server block can be easer tested in all envs:
- server_name can be for example redirect.etsin.csc.local (or whatever the local dev url pattern is for etsin).
- Similarly for test, stable etc envs, for example redirect.etsin-test.fairdata.fi. Certs for that url should use self-signed certifiactes (created by ansible, in the role "certificates") for other envs than production.
- The name of the used certificate should be parameterized (e.g. {{ old_etsin_cert_name }}, etc)
- The final redirect url should use the existing {{ server_domain_name }} in all envs, so that for example redirect.etsin-test.fairdata.fi redirects to etsin-test.fairdata.fi.

Doing these things is not a lot of work, and should be a good exercise in modifying ansible playbooks and conf files.

Reply:
Thanks for a thorough review and related instructions. I will improve and test the code based on these change requests.